### ISSUE TITLE
Widget: Notification & Sound each time I change AnkiDroid Activities

### ISSUE THREAD

**Issue by david-allison-1** [ankidroid/Anki-Android#6476]
2.12alpha6 Android 9.

I don't use notifications/widget generally, so this may be user error.

I have a test collection with 55 unseen cards

Notification settings: Notify when more than 10 cards due

Each time I lock my phone with AnkiDroid open, or load a new activity in AnkiDroid, I get a notification sound and notification with "55 cards due". Even if the notification is currently appearing, there's no grace period.

This is distracting and isn't a great user experience.

Caused by calling `widget.update()` `onStop`: https://github.com/ankidroid/Anki-Android/blob/b1135ff3cbd8d536a0175e7b9b156fdb88585e1b/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java#L683

This calls through to:

https://github.com/ankidroid/Anki-Android/blob/8ada06b2e6521aa4e9eff2a0539dc192ef89a894/AnkiDroid/src/main/java/com/ichi2/widget/WidgetStatus.java#L116

**Comment by mikehardy** [comment]
Yes it's pretty sucky at the moment. I have a "notifications refactor" branch alive on my fork but it had very little solid progress and wasn't as important as database corruption (5151) so has sat for nearly 2 years! I know a lot more about notifications now and would say that I think updating the notification (if it exists) as opposed to clearing and posting a new one is the correct way but everything about notifications in any app needs careful testing

**Comment by dae** [comment]
@david-allison still happening?
